### ISSUE TITLE
Corruption when upgrade from v4.0.9 to nightly

### ISSUE THREAD

**Issue by hunterlxt** [tikv/tikv#9701]
### What version of TiKV are you using?
master c514eb5e71bc0e150a168463d2fd7ad777fffaa7

### What operating system and CPU are you using?

DBaas dev

### Steps to reproduce
```
[security.encryption]
data-encryption-method = "aes128-ctr"
data-key-rotation-period = "7d"

[security.encryption.master-key]
type = "file"
path = "/root/key"
```

```
tiup cluster start <cluster-name>
tiup cluster upgrade <cluster-name> nightly
```

### What did you expect?
No crash

### What did happened?
```
[2021/02/20 05:21:06.243 +00:00] [ERROR] [server.rs:848] ["failed to init io snooper"] [err_code=KV:Unknown] [err="\"IO snooper is not started due to not compiling with BCC\""]
[2021/02/20 05:21:08.600 +00:00] [INFO] [mod.rs:96] ["encryption: found both of key dictionary and file dictionary."]
[2021/02/20 05:21:08.600 +00:00] [INFO] [mod.rs:496] ["encryption is enabled. method = Aes256Ctr"]
[2021/02/20 05:21:08.601 +00:00] [FATAL] [server.rs:952] ["failed to create raft engine: Storage Engine Corruption: CURRENT file does not end with newline"]
```

**Comment by hunterlxt** [comment]
@yiwu-arbug and I confirmed that the reason: TiKV 5.0 skips the encryption of the file `CURRENT`, but didn't take upgrade into account. After restarting TiKV to 5.0, plaintext `CURRENT` will be written. but this information is not deleted in `file.dict`. When next restart, TiKV will read a plaintext file as an encrypted file.

**Comment by hunterlxt** [comment]
this has been fixed in #9736
